src/charts/stacked-bar.js

```diff
@@ -0,0 +1,1080 @@
+                .attr('transform', `translate(${yTickTextXOffset}, ${yTickTextYOffset})`);
+        }
+
+         /**
+         * Draws grid lines on the background of the chart
+         * @return void
+         */
+        function drawGridLines(){
+            if (!horizontal) {
+                drawVerticalGridLines();
+            } else {
+                drawHorizontalGridLines();
+            }
+        }
+
+        /**
+         * Draws the grid lines for an horizontal bar chart
+         * @return {void}
+         */
+        function drawHorizontalGridLines() {
+            maskGridLines = svg.select('.grid-lines-group')
+                .selectAll('line.vertical-grid-line')
+                .data(xScale.ticks(4))
+                .enter()
+                  .append('line')
+                    .attr('class', 'vertical-grid-line')
+                    .attr('y1', (xAxisPadding.left))
+                    .attr('y2', chartHeight)
+                    .attr('x1', (d) => xScale(d))
+                    .attr('x2', (d) => xScale(d))
+
+            //draw a horizontal line to extend y-axis till the edges
+            baseLine = svg.select('.grid-lines-group')
+                .selectAll('line.extended-y-line')
+                .data([0])
+                .enter()
+                  .append('line')
+                    .attr('class', 'extended-y-line')
+                    .attr('y1', (xAxisPadding.left))
+                    .attr('y2', chartHeight)
+                    .attr('x1', 0)
+                    .attr('x2', 0);
+        }
+
+        /**
+         * Draws the grid lines for a vertical bar chart
+         * @return {void}
+         */
+        function drawVerticalGridLines() {
+            maskGridLines = svg.select('.grid-lines-group')
+                .selectAll('line.horizontal-grid-line')
+                .data(yScale.ticks(4))
+                .enter()
+                  .append('line')
+                    .attr('class', 'horizontal-grid-line')
+                    .attr('x1', (xAxisPadding.left))
+                    .attr('x2', chartWidth)
+                    .attr('y1', (d) => yScale(d))
+                    .attr('y2', (d) => yScale(d))
+
+            //draw a horizontal line to extend x-axis till the edges
+            baseLine = svg.select('.grid-lines-group')
+                .selectAll('line.extended-x-line')
+                .data([0])
+                .enter()
+                  .append('line')
+                    .attr('class', 'extended-x-line')
+                    .attr('x1', (xAxisPadding.left))
+                    .attr('x2', chartWidth)
+                    .attr('y1', height - margin.bottom - margin.top)
+                    .attr('y2', height - margin.bottom - margin.top);
+        }
+
+
+        /**
+         * Draws the bars along the x axis
+         * @param  {D3Selection} bars Selection of bars
+         * @return {void}
+         */
+        function drawHorizontalBars() {
```

Comment:
This would complicate things further, but now we are checking an 'animation' flag in order to differentiate when to add animatiions or not on the bar drawing.
It would be great if we could add that here too.